### PR TITLE
[ENG-320] Update Token-Scope Relationships

### PR DESCRIPTION
**Pull request by cslzchen**
## Ticket

https://openscience.atlassian.net/browse/ENG-320

## Purpose

Update Token-Scope Relationships. See [here](https://github.com/CenterForOpenScience/osf.io/pull/8766) for the model change on PAT.

## Changes

*  Fix style with `go fmt`.
* Rewrite queries to obtain scopes for a token.

## Side effects

Token without a scope will return a error.

## QA Notes

DevOQ: please test this locally with https://github.com/CenterForOpenScience/osf.io/pull/8766.

## Deployment Notes

- [x] Can be merged into develop ahead of https://github.com/CenterForOpenScience/osf.io/pull/8766.

- [ ] Must be released to master at the same time with https://github.com/CenterForOpenScience/osf.io/pull/8766.

- [ ] After merged to develop, change image from `fakecas:master` to `fakecas:develop` in [docker-compose.yml](https://github.com/CenterForOpenScience/osf.io/blob/develop/docker-compose.yml#L237) for local development.

### CODE REVIEW

File: views.go [hunk 1]
```diff
@@ -161,45 +162,79 @@ func ServiceValidate(c echo.Context) error {
...
+			JOIN osf_apioauth2scope
+		   		on osf_apioauth2personaltoken_scopes.apioauth2scope_id = osf_apioauth2scope.id
+		WHERE osf_apioauth2personaltoken.token_id = $1
+	`
+	rows, err := DatabaseConnection.Query(queryString, tokenId)
+	if err != nil {
+		if err != sql.ErrNoRows {
+			panic(err)
+		}
+		fmt.Printf("No scope is found for access token %s\n", tokenId)
```
Comment (cslzchen) [inline views.go:211]: Previously, token with no scope returns an empty scope list. Now, token with no scope returns 404.

File: views.go [hunk 2]
```diff
@@ -161,45 +162,79 @@ func ServiceValidate(c echo.Context) error {
...
+		WHERE osf_apioauth2personaltoken.token_id = $1
 	`
-	err := DatabaseConnection.QueryRow(queryString, tokenId).Scan(&result.Id, &result.Username, &result.GivenName, &result.FamilyName, &scopes)
+	err := DatabaseConnection.QueryRow(queryString, tokenId).Scan(&result.Id, &result.Username, &result.GivenName, &result.FamilyName)
 	if err != nil {
 		if err != sql.ErrNoRows {
 			panic(err)
 		}
-		fmt.Println("Access token", tokenId, "not found")
+		fmt.Printf("Access token %s not found\n", tokenId)
```
Comment (cslzchen) [inline views.go:190]: `fmt.Printf()` is way better than `fmt.Print()` and `fmt.Println()`.

File: views.go [hunk 3]
```diff
@@ -161,45 +162,79 @@ func ServiceValidate(c echo.Context) error {
...
+	queryString = `
+		SELECT DISTINCT osf_apioauth2scope.name
+		FROM osf_apioauth2personaltoken_scopes
+			JOIN osf_apioauth2personaltoken
+		   		on osf_apioauth2personaltoken_scopes.apioauth2personaltoken_id = osf_apioauth2personaltoken.id
+			JOIN osf_apioauth2scope
+		   		on osf_apioauth2personaltoken_scopes.apioauth2scope_id = osf_apioauth2scope.id
+		WHERE osf_apioauth2personaltoken.token_id = $1
+	`
+	rows, err := DatabaseConnection.Query(queryString, tokenId)
```
Comment (cslzchen) [inline views.go:206]: Unlike `QueryRow()`, `Query()` needs triple error handling ... first on query, then on each row scan, finally on finish ...

**Review by cslzchen — COMMENTED** [review]
LGTM. Merging soon. 🎆 

* cc @sloria Given that everyone is using `fakecas:master` for local development now, I can merge this to `develop` so you guys (and I) can test it locally with `fakecas` using the docker image at least.

* The indentation looks weird since they are tabs. Let's ignore it for now.

**Comment by sloria** [comment]
>  I can merge this to develop so you guys (and I) can test it locally with fakecas using the docker image at least.

@cslzchen Makes sense to me!

**Comment by cslzchen** [comment]
fyi, @sloria image `quay.io/centerforopenscience/fakecas:develop` ready to be pulled by `docker-compose`

**Pull request merged**